packages/blueprint_repository/lib/src/blueprint_repository.dart

```diff
@@ -1,21 +1,55 @@
-import 'entities/entities.dart';
+import 'package:blueprint_repository/src/entities/entities.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:stream_transform/stream_transform.dart';
+
+const _usersCollectionName = 'users';
 
 /// {@template blueprint_repository}
 /// Repository to handle blueprints on the platform.
 /// {@endtemplate}
 class TodaysBlueprintRepository {
   /// {@macro blueprint_repository}
-  const TodaysBlueprintRepository();
+  TodaysBlueprintRepository({
+    required Stream<String?> currentUserIdStream,
+    required FirebaseFirestore firestore,
+  }) {
+    _currentUserIdStream = currentUserIdStream;
+    _usersCollection = firestore.collection(_usersCollectionName);
+  }
+
+  late final Stream<String?> _currentUserIdStream;
+
+  late final CollectionReference _usersCollection;
 
-  /// Streams the todays blueprint from database.
+  /// Return a stream of the current user blueprints.
   Stream<List<CalendarEvent>> getBlueprint() {
-    // TODO(yarnr-p): implement getBlueprint
-    throw UnimplementedError();
+    return _currentUserIdStream.switchMap<List<CalendarEvent>>((userId) {
+      if (userId == null) {
+        return const Stream.empty();
+      }
+      final userDoc = _usersCollection.doc(userId);
+      return userDoc.snapshots().map((snapshot) {
+        final userData = snapshot.data() as Map<String, dynamic>?;
+        final blueprint = userData?['blueprint'] as List<dynamic>? ?? [];
+        return blueprint.map((e) {
```

Comment:
I don't think we need the Map<String,dynamic> check condition if we are throwing an exception.
```suggestion
return blueprint
            .map((e) => CalendarEvent.fromJson(e as Map<String, dynamic>))
            .toList();